lua/lualine/config.lua

```diff
@@ -6,6 +6,265 @@ local modules = require('lualine_require').lazy_require {
+---@field winbar table
+
+---@class LualineRefresh
+---@field statusline integer
+---@field tabline integer
+---@field winbar integer
+
+---@class LualineOpts
+-- Show icons.
+---@field icons_enabled boolean
+-- Lualine theme name.
+-- Auto theme will load the theme automaically for your colorscheme
+-- or will generate one otherwise.
+---@field theme LualineThemes
+-- Separators between components.
+---@field component_separators table|string
+-- Separators between sections.
+---@field section_separators table|string
+-- Disable lualine for these filetypes.
+---@field disabled_filetypes LualineDisabledFiletypes
+-- When set to true, left sections i.e. 'a','b' and 'c'
+-- can't take over the entire statusline even
+-- if neither of 'x', 'y' or 'z' are present.
+---@field always_divide_middle boolean
+---@field globalstatus boolean
+---@field refresh LualineRefresh
+
+---@alias LualineComponents
+---| "branch"
+---| "buffers"
+---| "diagnostics"
+---| "diff"
+---| "encoding"
+---| "fileformat"
+---| "filename"
+---| "filesize"
+---| "filetype"
+---| "hostname"
+---| "location"
+---| "mode"
+---| "progress`"
+---| "tabs"
+
+---@class LualineGeneralComponent
+-- Must be written as array value
+---@field [1] LualineComponents
+-- Enables the display of icons alongside the component.
+---@field icons_enabled boolean
+-- Defines the icon to be displayed in front of the component.
+---@field icon string
+-- When a string is provided it's treated as component_separator.
+-- When a table is provided it's treated as section_separator.
+-- Passing an empty string disables the separator.
+---@field separator string|table
+-- Condition function, the component is loaded when the function returns `true`.
+---@field cond fun():boolean
+-- Defines a custom color for the component.
+-- example: { fg = '#ffaa88', bg = 'grey', gui='italic,bold' } | { fg = 204 } | "WarningMsg"
+---@field color string|table
+-- Specify what type a component is.
+---@field type string|table
+-- Adds padding to the left and right of components.
+---@field padding number
+-- Format function, formats the component's output.
+---@field fmt fun():string
+
+---@class LualineColors
+-- Color for active buffer.
+--@field active string|table
+-- Color for inactive buffer.
+--@field inactive string|table
+
+---@class LualineBuffer: LualineGeneralComponent
+-- Shows shortened relative path when set to false.
+---@field show_filename_only boolean
+-- Shows indicator when the buffer is modified.
+---@field show_modified_status boolean
+-- 0: Shows buffer name
+-- 1: Shows buffer index (bufnr)
+-- 2: Shows buffer name + buffer index (bufnr)
+---@field mode number
+-- Maximum width of buffers component.
+---@field max_length number|fun():number
+-- Shows specific buffer name for that filetype.
+---@field filetype_names table
+-- Buffers colors
+---@field buffers_color LualineColors
+
+---@class LualineDiagnosticsColors
+-- Diagnostics' error color.
+---@field error string|table
```

Comment:
Because we have so many places that take same type of colors I we must make a lualine color type.

I think something like this will be appropiate
```lua
---@class LualineColorTbl
---@field fg stiring
---@field bg string
---coma sperated list of special properties of highlight such as bold, italic....
---@field gui string
---@alias LualineColor
---| LualineColorTbl
---| string
---| function(): LualineColorTbl | string
```

Then apply it to all sort of color options.